test/internal/sync/sync.go

```diff
@@ -0,0 +1,29 @@
+// This package holds common synchronization primitives used in test writing
+package sync
+
+import (
+	"sync"
+	"testing"
+)
+
+type LazyString struct {
+	once sync.Once
+	f    func() (string, error)
+	s    string
+	err  error
+}
+
+func NewLazyString(f func() (string, error)) *LazyString {
+	return &LazyString{f: f}
+}
+
+func (x *LazyString) String(tb testing.TB) string {
+	tb.Helper()
+	// since we cannot call test.Helper in the function once.Do uses to call f, avoid
+	// tests and panics within the once, and save it for the outside.
```

Comment:
I don't think I understand this comment

Reply:
basically, if we call `tb.Log` or `tb.Fail` from within the function that we pass to `once`, even if we call `tb.Helper`, the stack trace will still be messed up and attribute the log to the `sync` package, and not our code, because the function is called from within `once.Do`, which does not declare `tb.Helper`.
Its basically stack frame shenanigans.